main.tf

```diff
@@ -1,16 +1,140 @@
+module "lambda_package" {
+  source  = "matti/resource/shell"
+  version = "~>1.0.7"
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "1.0.0"
+  # I want to be using `npm run-script build` here,
+  # but for some reason I can't work out `node-lambda` fails in that context
+  command     = "node-lambda package -e production -x 'event.json build.env bin .github' -I lambci/lambda:build-nodejs12.x"
+  working_dir = "${path.module}/lambda"
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+  trigger = join("", values({
+    package_lock_json = filemd5("${path.module}/lambda/package-lock.json")
+    package_json      = filemd5("${path.module}/lambda/package.json")
+    app               = filemd5("${path.module}/lambda/index.js")
+  }))
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name_prefix = "${var.name}-"
+  description = "Role for ${var.name} to create secret"
+
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+  tags               = var.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+locals {
+  zipfile = "${path.module}/lambda/build/lambda-ssh-key-secret-production.zip"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_execution" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "secretsmanager_read_write" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
```

Comment:
this is unnecessarily broad

Reply:
TF creates the secret, so the Lambda can just be given access to PutSecretVersion against that.